bad/source/lib.rs

```diff
@@ -40,22 +41,33 @@ pub struct SyntaxTree {}
 
 pub fn get_source_text(source: ProgramSource) -> (PathBuf, String) {
 	let mut source_text: String = String::new();
-	let mut source_path: PathBuf = PathBuf::new();
+	let source_path: PathBuf;
 	match source {
 		ProgramSource::Path(pathbuf) => { source_path = pathbuf.clone(); source_text = std::fs::read_to_string(pathbuf.as_path()).unwrap() },
-		ProgramSource::File(mut file) => { source_path = "<file?handle?>".into(); file.read_to_string(&mut source_text); },
-		ProgramSource::Stdin(mut stdin) => { source_path = "<stdin>".into(); stdin.read_to_string(&mut source_text); }
+		ProgramSource::File(mut file) => { source_path = "<file?handle?>".into(); file.read_to_string(&mut source_text).unwrap(); },
+		ProgramSource::Stdin(mut stdin) => { source_path = "<stdin>".into(); stdin.read_to_string(&mut source_text).unwrap(); }
 	};
 	(source_path, source_text)
 }
 
-pub fn lex(source_path: PathBuf, source_text: String, config: &CompilationConfiguration) -> TokenList {
-	let mut list: TokenList = TokenList::new(source_path, source_text);
-	let logos_lex = Token::lexer(list.context.source()).spanned();
-	for tok_and_range in logos_lex {
+pub fn lex(input_path: PathBuf, input_source: String, _config: &CompilationConfiguration) -> TokenList {
+	let mut list: TokenList = TokenList::new(input_path, input_source);
+	let source : &str = list.context.source();
+	let source_path : &std::path::Path = list.context.path();
+	let spanned_tokens = TokenName::lexer(list.context.source()).spanned();
+	for tok_and_range in spanned_tokens {
 		match tok_and_range {
-			(Token::Error, _range) => panic!("unrecognized sequence of text during scanning/lexing!"),
-			(tok , _range) => list.tokens.push(tok)
+			(lex::TokenName::Error, range) => {
```

Comment:
Since I think this is the only "unusual" arm, it might make sense to go with an if let...

```rs
if let TokenName::Error = tok {
  // error
}
// Continue as before.
```

Reply:
Can do!